src/aiperf/dataset/dataset_manager.py

```
@@ -246,21 +271,53 @@ async def _generate_inputs_json_file(self) -> None:
+        http_client = AioHttpClient(timeout=Environment.DATASET.PUBLIC_DATASET_TIMEOUT)
+        try:
+            record = await http_client.get_request(
+                url, headers={"Accept": "application/json"}
+            )
+            dataset = record.responses[0].text
+            cache_filepath.parent.mkdir(parents=True, exist_ok=True)
+            cache_filepath.write_text(dataset)
+        except Exception as e:
+            raise DatasetLoaderError(
+                f"Error saving {tag} dataset to local cache: {e}"
+            ) from e
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)